### ISSUE TITLE
Slicer --launch option is ignored on MacOS

### ISSUE THREAD

**Issue by che85** [Slicer/Slicer#5540]
The `--launch` option is for some reason completely ignored without any error messages or indicator that an application was not found. 

Slicer will start normally without showing any error message

## Steps to reproduce

1. Install Slicer on MacOS, open the terminal and change to the folder where Slicer executable is located (e.g. /Applications/Slicer.app/Contents/MacOS/Slicer) 

2. run Slicer as follows:

```
./Slicer --launch
```

## Expected behavior

An error message is supposed to be displayed or the application argument after --launch should be executed

![image](https://user-images.githubusercontent.com/10195822/111798145-2a414e80-88a0-11eb-8067-3472e68a80b5.png)

## Environment
- Slicer version: 4.13.0-2021-03-12 r29771 / a3cf4cf
- OS version: macOS Catalina 10.15.7 (19H15)

**Comment by pieper** [comment]
Yes, this is true of packaged Slicer on macOS.  The launcher exists and works in a locally built version.

It would make sense to generate an error message, but even better could be to incorporate the launcher logic into the app itself.

A workaround for this would be to the the Slicer executable with the `--python-script` and `--no-main-window` options and make the script launch the subprocess using python or Qt.

**Comment by jcfr** [comment]
> but even better could be to incorporate the launcher logic into the app itself.

Agreed. An issue has been created to track this: https://github.com/commontk/AppLauncher/issues/122
